HW02_B_Mimic_starter.Rmd

```diff
@@ -1,99 +1,142 @@
+#theme_bw() is best theme (IMO)
+
+#for graph 3:
+library("ggrepel")
+```
+
+## HW02 Part B
+
+For this part of the HW, the goal is to try to recreate the graphs I make from scratch. I will only provide the MD, not the actual code I used to create it besides which data I use to create it. The rest will be up to you. 
+
+Try for all 4, but if you are struggling don't worry about it. Try your best for each, if you don't get everything that's what the peer-review is for. :smile:
+
+
+### Graph 1
+```{r, echo = T, include = TRUE}
+data("diamonds")
+#hint think about the *position* the bars are in...
+```
+
+
+Using the diamonds dataset, make this graph:
+```{r graph1 code, echo=FALSE}
+p1 <- ggplot(diamonds, aes(x = cut, fill = clarity)) +
+  # Make bars for each cut side by side
+  geom_bar(position = "dodge") +
+  # Add titles and axis labels
+  labs(title = "My Diamond Collection", subtitle = "Boxplot representing the number of diamonds in my diamond collection by \ntype of cut quality and clarity of diamond", x = "Diamond Cut", y = "Number of Diamonds") + 
+  # Center the titles
+  theme(plot.title = element_text(hjust = 0.5)) +
+  # Text annotation for diamonds with ideal cut
+  annotate("text", x = 4, y = 4500, label = "My Best Diamonds, \n of course") +
+  # Rectangular box to highlight diamonds with ideal cut 
+  annotate("rect", xmin = 4.5, xmax = 5.5, ymin = 0, ymax = 5000, alpha = 0.3)
+
+p1
+```
+
+### Graph 2
+```{r, echo = T, include = TRUE}
+data("iris")
+```
+
+Using the iris dataset, make this graph:
+```{r graph 2 code, echo=FALSE}
+p2 <- ggplot(iris, aes(x = Sepal.Length, y = Petal.Length, color = Species)) +
+  # Add the point layer to the plot, and specify shapes by species
+  geom_point(aes(shape = Species)) + 
+  # Wrap the plots for the three species to be three horizontally parallel graphs, and adjust the vertical scale for each plot 
+  facet_wrap(. ~Species, scales = "free_y") + 
+  # Add the regression lines, turn off the standard errors, change line color to black, and make lines to be thinner
+  geom_smooth(method = 'lm', se = FALSE, color = "black", size = 0.5)
+
+p2
+```
+
+
+### Graph 3
+You'll need the information in this first box to create the graph
+```{r, echo = TRUE}
+data("mpg")
+corvette <- mpg[mpg$model == "corvette",]
+#install
+require("ggrepel") #useful for making text annotations better, hint hint
+set.seed(42)
+```
+
+Now using the mpg dataset and the corvette dataset, make this graph:
+
+```{r graoh 3 code}
+p3 <- ggplot(mpg, aes(x = displ, y = hwy)) +
+  geom_point() +
+  # Add an extra layer of points using the corvette dataset
+  geom_point(aes(x = displ, y = hwy), corvette, color = "blue") +
+  # Add the plot title
+  labs(title = "Corvettes are a bit of an outlier") + 
+  # Bolden the plot title
+  theme(plot.title = element_text(face = "bold")) + 
+  # Add annotations for the corvette data points
+  geom_text_repel(aes(x = displ, y = hwy, label = paste0("Corvette, ", year)), corvette)
```

Comment:
I like the way geom_text_repel() function adds non-overlapping labels to data points elegantly. I hope SAS could developed a similar function...